src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -282,6 +282,12 @@ public void Load()
                     // Defaults
                     Options.ApplyHttpsDefaults(httpsOptions);
 
+                    // Default SslProtocol from config
+                    if (ConfigurationReader.EndpointDefaults.SslProtocols.HasValue)
+                    {
+                        httpsOptions.SslProtocols = ConfigurationReader.EndpointDefaults.SslProtocols.Value;
```

Comment:
HttpsConnectionAdapterOptions should be as fully populated from config ass possible before calling ApplyHttpsDefaults which calls into user code with the options object.

I'd remove the two if blocks that were added to this method and add the following right before ApplyHttpsDefaults.

```C#
httpsOptions.SslProtocols = endpoint.SslProtocols ?? ConfigurationReader.EndpointDefaults.SslProtocols ?? SslProtocols.None;
```

Reply:
We should also add a KestrelConfigurationLoaderTest that verifies ConfigureHttpsDefaults "sees" the SslProtocols that were defined in config.

Reply:
I was thinking something similar: Set EndpointDefaults.SslProtocols before calling ApplyHttpsDefaults. However, you don't to set an endpoint specific SslProtocols until later because you don't want that to be overriden by ApplyHttpsDefaults.

Reply:
> However, you don't to set an endpoint specific SslProtocols until later because you don't want that to be overriden by ApplyHttpsDefaults.

Very good point. We should have a test verifying that ApplyHttpsDefaults doesn't override SslProtocols configured directly on the enpoint.

Reply:
Updated and added unit tests for KestrelConfigurationLoader